### ISSUE TITLE
Sort desc first

### ISSUE THREAD

**Issue by jhonattan471** [akveo/ng2-smart-table#974]
Hello,

Would be very helpful if i can sort by 'desc' first, i have a table with many rows and i need to sort twice every time because 99% time need 'desc' sort ...

There is a native way, or i need to implement a custom sort?

**Comment by agupadilla** [comment]
Use "SortDirection" like this:

public settings = {
    actions: false,
    columns: {
      columnKey: {
        title: 'title', 
        **sortDirection: 'desc'**
      }
    }
  };

[documentation](https://akveo.github.io/ng2-smart-table/#/documentation)

**Comment by jhonattan471** [comment]
No... i want that when i click in the [sort] of the colunm, the first sort direction be 'DESC', today is 'ASC" by default... 
@agupadilla, This config only gives me a start sort when loading the table.

**Comment by gabrielvisconti89** [comment]
Any possible solution so far?

> No... i want that when i click in the [sort] of the colunm, the first sort direction be 'DESC', today is 'ASC" by default...
> @agupadilla, This config only gives me a start sort when loading the table.

I'm in the same situation, any solution so far?

**Comment by onyamegatron** [comment]
Quite a bit late but to answer OP of first column header click defaulting to desc.

```typescript
      foo: {
        title: 'foo',
        type: 'number',
        compareFunction(direction: any, a: any, b: any): number {
          if (a < b) {
            return -1 * direction
          } else if (a > b) {
            return direction
          }
          return 0;
        }
      },
``` 
Note you may have to parse a and b if you have non-number values.
